src/server/network/master.c

```
@@ -120,6 +120,48 @@ bool bc_master_add(bc_master_list_t *ml, const char *host_port, u16 game_port)
+{
+    if (!path || !out || max_out <= 0) return -1;
+
+    FILE *f = fopen(path, "r");
+    if (!f) return -1;
+
+    int count = 0;
+    char line[256];
+    while (count < max_out && fgets(line, sizeof(line), f)) {
+        /* Strip trailing newline / carriage return. */
+        size_t len = strlen(line);
+        while (len > 0 && (line[len - 1] == '\n' || line[len - 1] == '\r'))
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)